SU2_CFD/src/solvers/CTurbSASolver.cpp

```diff
@@ -289,6 +286,24 @@ void CTurbSASolver::Postprocessing(CGeometry *geometry, CSolver **solver_contain
   AD::EndNoSharedReading();
 }
 
+void CTurbSASolver::Viscous_Residual(unsigned long iEdge, CGeometry* geometry, CSolver** solver_container,
+                                     CNumerics* numerics, CConfig* config) {
+
+  /*--- Define an object to set solver specific numerics contribution. ---*/
+  struct SolverSpecificNumerics {
+
+    FORCEINLINE void operator() (const CGeometry& geometry, CNumerics& numerics, CConfig& config, const CScalarVariable& nodes, unsigned long iPoint, unsigned long jPoint) const {
+      /*--- Roughness heights. ---*/
+      if (config.GetKind_Turb_Model() == SA)
+        numerics.SetRoughness(geometry.nodes->GetRoughnessHeight(iPoint), geometry.nodes->GetRoughnessHeight(jPoint));
+    }
+
+  } SolverSpecificNumerics;
```

Comment:
e.g. if you reduce the signature to iPoint/jPoint, also note that you would need to template the "nodes" of the struct for when you template the scalar solver. If you let the caller "capture" them that is not necessary.
```suggestion
  auto SolverSpecificNumerics = [&](unsigned long iPoint, unsigned long jPoint) {
    if (config->GetKind_Turb_Model() == SA)
      numerics->SetRoughness(geometry->nodes->GetRoughnessHeight(iPoint),
                             geometry->nodes->GetRoughnessHeight(jPoint));
  };
```

Reply:
👍 ok so a lambda function instead of struct. That is indeed nicer to just capture everything that is used inside by reference

Reply:
yep, if all you need is "operator ()", lambdas are simpler I think.